File: find_toUpperCase.js
const fs = require('fs');
const path = require('path');

const results = [];

function walk(dir) {
  if (results.length >= 20) return;
  const items = fs.readdirSync(dir);
  for (const item of items) {
    if (results.length >= 20) return;
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      walk(fullPath);
    } else if (item.endsWith('.js')) {
      try {
        const content = fs.readFileSync(fullPath, 'utf-8');
        const matches = content.match(/\.(toUpperCase)\(\)/g);
        if (matches) {
          for (const match of matches) {
            const idx = content.indexOf(match);
            const context = content.substring(Math.max(0, idx - 100), Math.min(content.length, idx + 50));
            if (!context.includes('typeof') && !context.includes('=== ') && !context.includes('string')) {
              results.push(`${fullPath}: ...${context}...`);
              if (results.length >= 20) return;
            }
          }
        }
      } catch (e) {}
    }
  }
}

walk('/app/node_modules');
for (const r of results) {
  console.log(r);
}

File: find_toUpperCase.py
import os, re

results = []
for root, dirs, files in os.walk('/app/node_modules'):
    for file in files:
        if file.endswith('.js'):
            path = os.path.join(root, file)
            try:
                with open(path, 'r', encoding='utf-8', errors='ignore') as f:
                    content = f.read()
                for match in re.finditer(r'\.toUpperCase\(\)', content):
                    # Get some context
                    start = max(0, match.start() - 100)
                    end = min(len(content), match.end() + 50)
                    context = content[start:end]
                    if 'typeof' not in context and '=== ' not in context and 'string' not in context:
                        results.append(f"{path}: ...{context}...")
                        if len(results) >= 20:
                            break
            except:
                pass
        if len(results) >= 20:
            break
    if len(results) >= 20:
        break

for r in results:
    print(r)
